### ISSUE TITLE
Global search doesn't show results but folder search does

### ISSUE THREAD

**Issue by bloppernd123** [Betterbird/thunderbird-patches#385]
**Issue:**
I am using yahoo mail with over 30K messages (though none of the folders have more than 10k) emails. All emails if less than 50Kb are stored locally stored. I know that there's a message adressed to loremipsum@yahoo.com in the unclassified folder. So if Search "loremipsum" in the folder search bar, the message is shown immediately. But when I use "Global search", i.e. the one on the top of the betterbird, it doesn't show that email :(

I have rebuilt the globaldatabase by deleting the global-messages-db.sqlite as mentioned here: https://support.mozilla.org/en-US/kb/rebuilding-global-database, but it doesn't seem to help. I have also compacted the database, but ti doesn't help. In general I find that global search doesn't show all the results. Please see the screenshot to understand. Probably expression-serach ng addon might show relevant results, but I can't analyse or see the code to understand what data it might send outside my pc.

Other supporting info as mentioned on the bug template:
1. OS: Windows 11 (All updates installed)
2. Anti-virus program: Windows defender
3. RAM: 32GB
4. Version of Betterbird: 128.5.2esr-bb19 (64-bit)
5. Account: 1 Yahoo account via IMAP
6. State whether the folder producing the problem is synchronised for offline use or not: Yes 
7. For all account types: How many messages? Between 30-40k
8. How big is the folder in KB, MB or GB: Profile folder is 1.3 GB (Free space on HD is 250GB)
9.  State whether you have configured any online services, like CalDAV calendars or CardDAV address books: No
10.  Check whether the problem reproduces with all add-ons switched off: Doesn't help
11. Have you manipulated any preferences, perhaps via user.js?: No
12. For performance issues, submit a (startup) performance profile: N/A
13. Does the issue happen in Thunderbird when run on the same profile?: Didn't check
14. For "can't send" or "can't receive" problems, disable your anti-virus program: N/A
15. Profiles in Dropbox folders don't work at all: N/A

![bug](https://github.com/user-attachments/assets/636125af-5c9a-4324-afc8-1413c36ea2dd)

**Comment by Betterbird** [comment]
Thanks for getting in touch. So nice that someone actually read the section about what information to provide and provided it.

My reply will be very sad in light of all this effort, in addition to the fact that around Christmas time one expects people to be friendly, helpful and nice.

We appreciate that you realise that there are two very distinct searches in TB/BB: There is Global Search (Gloda, for Global Database) and there is "regular search", the one that is used for the Quick Filter (QF) and the "Search Messages" panel, and it's also used when evaluating filters or for "saved searched" (virtual folders).

The latter normal search is fully maintained, and BB added quite a few enhancements: Regular expressions, complex search terms, ignoring diacritics, and there are some tweaks to the QFB for negations, tags, feedback, etc.

The former search, Gloda, is not maintained at all. It was once written by an ingenious developer, so ingenious that when he (it was a he) left the project, absolutely no one had the faintest clue that the Goda module was doing. There may have been some refactoring in that module, I believe the unit tests were redone, but the code is essentially the same. The BB project has added two modifications: You can index encrypted messages, and you can add the results as a list immediately. Both modifications were rather painful, the immediate list is also hacky, since it was glued onto a module we also have very little understanding of.

In other words: If it doesn't work, it won't be fixed. That said, I'm pretty sure that the global database works fine on basic cases, that is no fancy non-English text, as long as you don't move messages around too much, compact or repair folders, etc.

I have the feeling that some folders aren't indexed. That would apply to Trash and Junk folders, I'm not sure whether Yahoo "Unclassified" is some sort of Junk folder. Have you looked in the folder properties, here a screenshot from a Trash folder:

![image](https://github.com/user-attachments/assets/a855f8d7-19e5-4889-8d4b-899459672325)

In summary: We won't action it.

**Comment by bloppernd123** [comment]
Wow, that's a super fast reply considering a Sunday in the Christmas vacation period :)

Thanks for providing an elaborate answer. The unclassified is a custom folder that I have created and yes it's included in global search results, that's why it's showing 3 results in the screenshot from the same unclassified folder, but not the one that I expected.

I have actually compacted, deleted, moved too many files and folders..because of stupid yahoo limitation that more than 10k messages can't be synced. In the hindsight, I should haven't created yahoo account at the first place, but that's another story.

I didn't know that quick filter also supports regex etc..that would be a great feature to test, but anyways now I know that Gloda is broken..and the mystery will remain unsolved, until someone really looks into search module from scratch (some inspirations from Mailstore project may be). I suspect that expression-NG addon might not work as well, if it looks at the same index created by Gloda module.

Anyways many thanks you for your reply and wish you happy holidays and a merry Christmas.

**Comment by Betterbird** [comment]
You misunderstood: We added regular expressions for "Search Messages".
